src/js/actions/bindChannel/parseGuides.ts

```diff
@@ -1,22 +1,22 @@
-'use strict';
-
-var merge = require('lodash.merge'),
-    addGuide = require('../guideActions').addGuide,
-    actions = require('./helperActions'),
-    addAxisToGroup = actions.addAxisToGroup,
-    addLegendToGroup = actions.addLegendToGroup,
-    Guide = require('../../store/factory/Guide'),
-    imutils = require('../../util/immutable-utils'),
-    getIn = imutils.getIn,
-    getInVis = imutils.getInVis;
-
-var GuideType = Guide.GuideType,
-    CTYPE = {
-      x: GuideType.Axis, y: GuideType.Axis,
-      color: GuideType.Legend, size: GuideType.Legend, shape: GuideType.Legend
-    };
-
-var SWAP_ORIENT = {
+import {Guide, GuideType, AxisRecord, LegendRecord} from '../../store/factory/Guide';
+import {addGuide} from '../guideActions';
+import {Dispatch} from 'redux';
+import {State} from '../../store';
+
+
+const actions = require('./helperActions'),
+  addAxisToGroup = actions.addAxisToGroup,
+  addLegendToGroup = actions.addLegendToGroup,
+  imutils = require('../../util/immutable-utils'),
+  getIn = imutils.getIn,
+  getInVis = imutils.getInVis;
+
+const CTYPE = {
```

Comment:
I would try to replace this with if statements and type assertions, since it looks like that's what this dictionary is for (mapping the presence of properties to the guide type)